c8d/history: Fix non-native platforms

vvoland opened the pull request:
- fixes: https://github.com/moby/moby/issues/50851

When building a non-native platform, it's not unpacked by default.
History tries to read the disk usage of all the layer and it doesn't handle missing snapshots gracefully. 
This patch fixes this.        


```markdown changelog
containerd image store: Fix `docker history` failing with `snapshot X does not exist` when calling on a non-native image that was built locally
```

vvoland committed:
integration/internal: Handle Buildkit in GetImageIDFromBody

BuildKit emits some additional events during build and they are not
`build.Result` so don't fail if we encounter one.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
c8d/history: Fix non-native platforms

When building a non-native platform, it's not unpacked by default.
History tries to read the disk usage of all the layer and it doesn't
handle missing snapshots gracefully.

This patch fixes this.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented on integration/internal/build/build.go:
```diff
@@ -43,10 +41,18 @@ func GetImageIDFromBody(t *testing.T, body io.Reader) string {
+			break
+		}
 	}
 	io.Copy(io.Discard, body)
```
If you still need to update; while we're updating; perhaps also explicitly discard the error;

```go
	_, _ = io.Copy(io.Discard, body)
```

thaJeztah reviewed (approved):
LGTM

vvoland committed:
integration/internal: Print Buildkit logs

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
gha/arm64: Setup qemu

Make it possible to build non-native images like the other
test-integration workflows

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

austinvazquez reviewed (approved):
LGTM
